backend/src/connectors/store.py

```
@@ -329,6 +358,33 @@ def flush(reason="batch"):
+            with DOWNLOAD_LOCK:
+                file_path = file.download(temp_path)
+
+            index = TreeDex.from_file(str(file_path), llm=llm, extract_images=get_bool_env('LONG_CONTEXT_IMGS'))
+            index.save(index_path)
+
+            os.remove(file_path)
+            
+        else:
+            logging.info(f'Index for {file.metadata["id"]} already found')
+
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)